packages/dds/tree/src/shared-tree/treeAlpha.ts

```diff
@@ -339,6 +353,44 @@ export interface TreeAlpha {
 	 * Otherwise, this returns the key of the field that it is under (a `string`).
 	 */
 	key2(node: TreeNode): string | number | undefined;
+
+	/**
+	 * Gets the child of the given node with the given property key if a child exists under that key.
+	 *
+	 * @remarks Unknown optional fields of Object nodes cannot be accessed using this method.
+	 *
+	 * @param node - The parent node whose child is being requested.
+	 * @param key - The property key under the node under which the child is being requested.
+	 * Note: this is the "property key", not the "stored key".
+	 *
+	 * @returns The child node or leaf value under the given key, or `undefined` if no such child exists.
+	 *
+	 * @see {@link (TreeAlpha:interface).key2}
+	 * @see {@link (TreeNodeApi:interface).parent}
+	 */
+	child(node: TreeNode, key: string | number): TreeNode | TreeLeafValue | undefined;
+
+	/**
+	 * Gets the children of the provided node, paired with their property keys under the node.
+	 *
+	 * @remarks
+	 * No guarantees are made regarding the order of the children in the returned array.
+	 *
+	 * Unset optional properties of Object nodes are not included in the result.
+	 *
+	 * Unknown optional fields of Object nodes are not included in the result.
```

Comment:
By "unknown" does this mean not-in-the-schema? Since that's a thing that can only happen under specific circumstances (you need the flag), can we have this sentence link to the `allowUnknownOptionalFields` API? Otherwise, this statement will likely be non-sensical to somebody who doesn't know about that feature ("what's an unknown field?").

Reply:
Same for the above docs on `child`